src/coreclr/vm/jithelpers.cpp

```diff
@@ -5375,13 +5387,15 @@ void JIT_PartialCompilationPatchpoint(int ilOffset)
     // Install new entry point as IP
     SetIP(&frameContext, osrMethodCode);
 
-    // Restore last error (since call below does not return)
-    // END_PRESERVE_LAST_ERROR;
+    // This method doesn't return normally so we have to manually restore things.
+    HELPER_METHOD_FRAME_END();
     ::SetLastError(dwLastError);
+    ENDFORBIDGC();
```

Comment:
`ENDFORBIDGC` may need to be before `SetLastError`.

Reply:
Thanks -- will reorder these.